hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ReplicationManager.java

```diff
@@ -543,11 +543,15 @@ private void handleUnderReplicatedContainer(final ContainerInfo container,
         List<DatanodeDetails> targetReplicas = new ArrayList<>(source);
         // Then add any pending additions
         targetReplicas.addAll(replicationInFlight);
-
-        int delta = replicationFactor - getReplicaCount(id, replicas);
         final ContainerPlacementStatus placementStatus =
             containerPlacement.validateContainerPlacement(
                 targetReplicas, replicationFactor);
+        int delta = replicationFactor - getReplicaCount(id, replicas);
```

Comment:
NIT: this line to move the line to calculate delta from 547->549 is not needed.  Only the line 550 is the core change that breaks out when we've already handle the under replicate container via inflight replication.

Reply:
This delta calculation and check in line 550 is required.  Currenlty placementStatus.isPolicySatisfied() only checks whether topology requirement is satisfied or not. It doesn't check whether it has enough replicas.

Reply:
Got it. I agree delta calculation is needed. I mean move delta calculation after placementStatus is not necessary.